pandas/_libs/tslibs/strptime.pyx

```diff
@@ -166,6 +173,9 @@ def array_strptime(
                 check_dts_bounds(&dts)
             result_timezone[i] = val.tzinfo
             continue
+        elif is_datetime64_object(val):
+            iresult[i] = get_datetime64_nanos(val, NPY_FR_ns)
+            continue
```

Comment:
I'll planning to do more code sharing between the ISO and non-ISO paths, but for now this is a minimal fix